File: js/pob.js
class Block {
    constructor(data, previousHash, burnedCoins) {
        this.data = data;
        this.previousHash = previousHash;
        this.burnedCoins = burnedCoins;
        this.hash = this.calculateHash();
    }

    calculateHash() {
        return CryptoJS.SHA256(this.data + this.previousHash + this.burnedCoins).toString();
    }
}

class Blockchain {
    constructor() {
        this.chain = [this.createGenesisBlock()];
        this.burnedCoins = 0;
    }

    createGenesisBlock() {
        return new Block("Genesis Block", "0", 0);
    }

    getLatestBlock() {
        return this.chain[this.chain.length - 1];
    }

    addBlock(newBlock) {
        newBlock.previousHash = this.getLatestBlock().hash;
        this.burnedCoins += newBlock.burnedCoins;
        this.chain.push(newBlock);
    }

    isChainValid() {
        for (let i = 1; i < this.chain.length; i++) {
            const currentBlock = this.chain[i];
            const previousBlock = this.chain[i - 1];

            if (currentBlock.hash !== currentBlock.calculateHash()) {
                return false;
            }

            if (currentBlock.previousHash !== previousBlock.hash) {
                return false;
            }
        }
        return true;
    }
}

File: swift/pob.swift
struct Block {
    var data: String
    var previousHash: String
    var miner: String
    var burnFee: Double
    var hash: String

    init(data: String, previousHash: String, miner: String, burnFee: Double) {
        self.data = data
        self.previousHash = previousHash
        self.miner = miner
        self.burnFee = burnFee
        self.hash = calculateHash()
    }

    func calculateHash() -> String {
        let dataString = data + previousHash + miner + String(burnFee)
        let dataData = dataString.data(using: .utf8)!
        let hash = SHA256.hash(data: dataData)
        return hash.map { String(format: "%02x", $0) }.joined()
    }
}

class Blockchain {
    var chain: [Block]
    var blockTime: Double
    var burnFees: [String: Double]

    init() {
        chain = [createGenesisBlock()]
        blockTime = 5000
        burnFees = [:]
    }

    func createGenesisBlock() -> Block {
        return Block(data: "Genesis Block", previousHash: "0", miner: "", burnFee: 0)
    }

    func getLatestBlock() -> Block {
        return chain[chain.count - 1]
    }

    func addBlock(newBlock: Block) {
        newBlock.previousHash = getLatestBlock().hash
        chain.append(newBlock)
    }

    func isChainValid() -> Bool {
        for i in 1..<chain.count {
            let currentBlock = chain[i]
            let previousBlock = chain[i - 1]

            if currentBlock.hash != currentBlock.calculateHash() {
                return false
            }

            if currentBlock.previousHash != previousBlock.hash {
                return false
            }
        }
        return true
    }

    func addBurnFee(miner: String, burnFee: Double) {
        burnFees[miner] = burnFee
    }

    func selectMiner() -> String {
        let highestBurnFee = burnFees.values.max()
        return burnFees.first(where: { $0.value == highestBurnFee })!.key
    }

    func mineBlock(block: Block) -> Bool {
        return block.miner == selectMiner()
    }
}
